health_check/views.py

```diff
@@ -126,8 +126,7 @@ def _run(plugin):
                 return self.render_to_response(context, status=status_code)
             elif media.mime_type in ('application/json', 'application/*'):
                 return self.render_to_response_json(plugins, status_code)
-            else:
-                return self.render_to_response(context, status=status_code)
+        return self.render_to_response(context, status=status_code)
```

Comment:
How can that ever happen if the default is `*/*` see a couple lines above?
The only way this would happen, is if someone sends an empty accept header. In this case it might probably sense to respond with a 400.

Reply:
The Accept header can be absent, or could just contain types that we don't care about in the conditions above.

There's no requirement, as far as I remember, that the caller includes `*/*` in their list, so I think it's legitimate to fall off the bottom of the loop.

Reply:
I would honestly like to see a test that breaks it and I still believe in this case we should respond with a 400, since we can't serve a response that would satisfy the request.

Reply:
Apologies. I thought I replied to this yesterday.

The old code did return HTML if the 'top' result from the request's accept list didn't match the explicit lists, so my change here was trying to keep that behaviour, but keep looking at the rest of the acceptable types.

I think you're right, though: it's reasonable to make that case a 400-series response. I think a 406 would be appropriate. I'll add a covering test and change the behaviour.

Reply:
@ashokdelphia, OK, go for it!

Reply:
Sorry for the delay responding.

I pushed up the changes.

I'm a little concerned this would be a breaking change for anyone currently hitting the endpoint with an unrecognized Accept header, but if you think that's okay, then feel free to merge.